File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
//import java.util.Hashset;
class Solution {
    public int lengthOfLongestSubstring(String s) 
    {
       HashSet<Character> set = new HashSet<>();

        int left = 0;
        int maxLength = 0;
        //loop through char
        for (int right = 0; right < s.length(); right++) {
            //update left so that there are no duplicates
            while (set.contains(s.charAt(right))) {
                set.remove(s.charAt(left));
                left++;
            }
            //add this to hashSet
            set.add(s.charAt(right));
            //update max length
            maxLength = Math.max(maxLength, right - left + 1);
        }

        return maxLength;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) 
    {
        int area = 0;
        int left = 0;
        int right = heights.size()-1;
        while (left < right)
        {
            int new_area = (right - left) * min(heights.at(left), heights.at(right));
            area = max(area, new_area);
            if(heights.at(left) < heights.at(right))
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        return area;
    }
};
